htdocs/stripe/class/stripe.class.php

```diff
@@ -425,19 +430,26 @@ public function createPaymentStripe($amount, $currency, $origin, $item, $source,
 					$fee = round($conf->global->STRIPE_APPLICATION_FEE_MINIMAL * 100);
 				}
 
-				$charge = \Stripe\Charge::create(array(
-					"amount" => "$stripeamount",
-					"currency" => "$currency",
-					// "statement_descriptor" => " ",
-					"description" => "$description",
-					"metadata" => $metadata,
-					"source" => "$source",
-					"customer" => "$customer",
-					"application_fee" => "$fee"
-					), array(
-					"idempotency_key" => "$ref",
-					"stripe_account" => "$account"
-				));
+        		$paymentarray = array(
+						"amount" => "$stripeamount",
+						"currency" => "$currency",
+                                                "capture"  => true,            
+						"statement_descriptor" => dol_trunc(dol_trunc(dol_string_unaccent($customer->name), 6, 'right', 'UTF-8', 1).' '.$description, 22, 'right', 'UTF-8', 1),     // 22 chars that appears on bank receipt
+					        "description" => "Stripe payment: ".$description,
+						"metadata" => $metadata,
+						"source" => "$source",
+						"customer" => "$customer"
+					);
+					if ($conf->entity>1 && $fee>0)
```

Comment:
Why using conf->entity > 1 ?
When using multicompany, company 1 is a company like other so like the company 2 or 3, so we should not have code that differs if we are company 1 or 2.

Reply:
I define a new constant for define principal entity (principal entity can't have application fee without error of API)